projects/packages/my-jetpack/src/products/class-backup.php

```diff
@@ -200,6 +205,90 @@ private static function get_state_from_wpcom() {
 		return $status;
 	}
 
+	/**
+	 * Hits the wpcom api to retrieve the last 10 backup records.
+	 *
+	 * @return Object|WP_Error
+	 */
+	public static function get_latest_backups() {
+		static $backups = null;
+
+		if ( $backups !== null ) {
+			return $backups;
+		}
+
+		$site_id  = \Jetpack_Options::get_option( 'id' );
+		$response = Client::wpcom_json_api_request_as_blog(
+			sprintf( '/sites/%d/rewind/backups', $site_id ) . '?force=wpcom',
+			'2',
+			array( 'timeout' => 2 ),
+			null,
+			'wpcom'
+		);
+
+		if ( 200 !== wp_remote_retrieve_response_code( $response ) ) {
+			$backups = new WP_Error( 'rewind_backups_fetch_failed' );
+			return $backups;
+		}
+
+		$body    = wp_remote_retrieve_body( $response );
+		$backups = json_decode( $body );
+		return $backups;
+	}
+
+	/**
+	 * Determines whether the module/plugin/product needs the users attention.
+	 * Typically due to some sort of error where user troubleshooting is needed.
+	 *
+	 * @return boolean|array
+	 */
+	public static function does_module_need_attention() {
+		$backup_failed_status = false;
+		// First check the status of Rewind for failure.
+		$rewind_state = self::get_state_from_wpcom();
+		if ( ! is_wp_error( $rewind_state ) ) {
+			$backup_failure_reasons = array(
```

Comment:
Are these all the possible failure statuses for backups that we know of? Will we have to add more to the array here if more are added?

Reply:
Yes these failure statuses come from the VaultPress/Rewind API. Looking through the code, these are all the possible backup error responses.
Yes if the VaultPress/Rewind API code was updated/changed for some reason and some other error code(s) were added for whatever reason, we would probably want to add it here too.  But I feel like the chances of that are very very slim. I think all these error codes have remained unchanged since the VaultPress/Rewind API was first created.

Reply:
Awesome thank you for clarifying 😄